pkg/wharfyml/nodemapparser.go

```diff
@@ -147,3 +176,49 @@ func (p nodeMapParser) newRequiredError(key string) error {
 	inner := fmt.Errorf("%w: %q", ErrMissingRequired, key)
 	return wrapPosErrorNode(inner, p.parent)
 }
+
+func (p nodeMapParser) readFromVarSubForOther(
```

Comment:
The bulk of this method is a duplicate of `lookupFromVarSubForOther`.

With the slight difference of saying `"%w: need %varLookup% or %nodeKey% to construct %other%"` for the first error instead of `"%w: need %varLookup% to construct %other%"`, line #185 to #197 is the same as `lookupFromVarSubForOther`.

Would be nice if the error could remain as-is but we could call `lookupFromVarSubForOther` in this method.

Maybe something like ( Diff here: https://www.textcompare.org/?id=6233241f197ed215ecce4900 )
```go
func (p nodeMapParser) readFromVarSubForOther(
	nodeKey, varLookup, other string, source varsub.Source) error {
	if _, ok := p.nodes[nodeKey]; ok {
		return nil
	}
	newNode, err := p.lookupFromVarSubForOther(varLookup, other, source, "or", nodeKey)
	p.nodes["registry"] = newNode
	return err
}

func (p nodeMapParser) lookupFromVarSubForOther(
	varLookup, other string, source varsub.Source, additionalErrs ...string) (*yaml.Node, error) {
	value, ok := safeLookupVar(source, varLookup)
	if !ok {
		err := fmt.Errorf("%w: need %s to construct %q",
			ErrMissingBuiltinVar, strings.Join(append([]string{varLookup}, additionalErrs...), " "), other)
		return nil, wrapPosErrorNode(err, p.parent)
	}
	newNode, err := newNodeWithValue(p.parent, value)
	if err != nil {
		err := fmt.Errorf("read %s to construct %q: %w",
			varLookup, other, err)
		return nil, wrapPosErrorNode(err, p.parent)
	}
	return newNode, nil
}
```
?

Not the best, but it reads a lot better IMO.

Reply:
Fixed!

I blame myself for coding while I was so tired.